### ISSUE TITLE
TestCase 65 212. Word Search II

### ISSUE THREAD

**Issue by ManavSarkar** [LeetCode-Feedback/LeetCode-Feedback#14925]
#### Your LeetCode username
manavsarkar07 

#### Category of the bug
- [ ] Question
- [ ] Solution
- [ ] Language
- [1] Missing Test Cases 

#### Description of the bug
The last testcase 65 has nothing. It is showing time limit exceed. 

#### Code you used for Submit/Run operation
```
class TrieNode {
public:
    unordered_map<char, TrieNode*> children;
    bool isEndOfWord;

    TrieNode() {
        isEndOfWord = false;
    }
};

class Trie {
private:
    TrieNode* root;

public:
    Trie() {
        root = new TrieNode();
    }

    void insert(string word) {
        TrieNode* current = root;
        for (char ch : word) {
            if (current->children.find(ch) == current->children.end()) {
                current->children[ch] = new TrieNode();
            }
            current = current->children[ch];
        }
        current->isEndOfWord = true;
    }

    bool search(string word) {
        TrieNode* current = root;
        for (char ch : word) {
            if (current->children.find(ch) == current->children.end()) {
                return false;
            }
            current = current->children[ch];
        }
        return current->isEndOfWord;
    }

    bool startsWith(string prefix) {
        TrieNode* current = root;
        for (char ch : prefix) {
            if (current->children.find(ch) == current->children.end()) {
                return false;
            }
            current = current->children[ch];
        }
        return true;
    }

    ~Trie() {
        clear(root);
    }

private:
    void clear(TrieNode* node) {
        if (!node) return;
        for (auto& pair : node->children) {
            clear(pair.second);
        }
        delete node;
    }
};
class Solution {
public:

    vector<string> ans;
    TrieNode* root;
    void insert(string word) {
        TrieNode* current = root;
        for (char ch : word) {
            if (current->children.find(ch) == current->children.end()) {
                current->children[ch] = new TrieNode();
            }
            current = current->children[ch];
        }
        current->isEndOfWord = true;
    }
    void dfs(string& curr, TrieNode* tr, int i, int j, vector<vector<char>> &board){
        if(i<0 || j<0 || i==board.size() || j==board[0].size() || board[i][j]=='.'){
            return;
        }
        
        char ch = board[i][j];
        curr.push_back(ch);
        board[i][j] = '.';
        if(tr->children.find(ch) != tr->children.end()){
            // cout<<curr<<endl;
            if(tr->children[ch]->isEndOfWord){
                ans.push_back(curr);
            }
            TrieNode* temp = tr->children[ch];
            dfs(curr,temp,i+1,j, board);
            dfs(curr,temp,i-1,j, board);
            dfs(curr,temp,i,j+1, board);
            dfs(curr,temp,i,j-1, board);
            // dfs(curr,temp,i+1,j+1, board);
            // dfs(curr,temp,i+1,j-1, board);
            // dfs(curr,temp,i-1,j+1, board);
            // dfs(curr,temp,i-1,j-1, board);
            
        }
        curr.pop_back();
        board[i][j] = ch;
    }
	vector<string> findWords(vector<vector<char> >& board, vector<string>& dictionary) {
	    Trie* trie = new Trie();
	    root = new TrieNode();
	    for(auto str:dictionary){
	        insert(str);
	    }
	    ans.clear();

	    for(int i=0;i<board.size();i++){
	        for(int j=0;j<board[0].size();j++){
                string s = "";
	            dfs(s,root,i,j,board);}
	    }
	    set<string> s( ans.begin(), ans.end() );
        ans.assign( s.begin(), s.end() );
	    return ans;
	}
};
```

#### Language used for code
C++

#### Expected behavior
It should have some test case

#### Screenshots

![image](https://github.com/LeetCode-Feedback/LeetCode-Feedback/assets/42129636/853bceea-288f-4c00-b518-195ee7cf57ee)

**Comment by LC-Pam** [comment]
Thank you for reaching out to us, the same code could TLE or AC because it's at the verge of not passing. This could occur because the code is not optimized. When you run your code by "Submit", it's very different from running it with "Run Code". "Submit" run all of the test cases for this problem at once, whereas "Run Code" only run one test case at a time. Furthermore, getting Time Limit Exceeded from "Submit" does not mean that your code is not passing a certain test case; it simply means that your code is taking too long to run all of the test cases at once. I found a Swift solution in the problem discuss section and hope it can provide some inspiration to how your code can be improved to overcome this TLE issue: https://leetcode.com/problems/elimination-game/discuss/245342/Swift-keep-left-and-right-boundary

The issue you experienced is runtime variation. It normally happens when your code is at the edge of the time limit that can pass the problem. Therefore, your code may have different results at different submissions.
 It’s a very difficult and ongoing research to evaluate the program’s runtime and get the same value for different runs of the same program. We have tried our best to make sure we isolate your program’s runtime, but there exist a lot of factors that make different executions have runtime variation.
 
I've passed your feedback to our engineering team and they'll investigate in order to provide consistent results for our users. However, I'm afraid that we won't have a quick fix for the issue that you're facing. I recommend that you optimize your code to pass the time limit. We apologize for the inconvenience and thank you for your support!
